From c5cebac8520ac6e08021a5708de31cc45daca2a3 Mon Sep 17 00:00:00 2001
From: Daniel <shatter@gmail.com>
Date: Wed, 7 Oct 2009 22:01:34 -0400
Subject: [PATCH] Remove svg requirement from cairo and add nosvg option to work around circular dependencies.


Signed-off-by: Daniel <shatter@gmail.com>
---
 packages/x11-libs/gtk+/gtk+-2.18.2.exheres-0 |   11 ++++++++++-
 1 files changed, 10 insertions(+), 1 deletions(-)

diff --git a/packages/x11-libs/gtk+/gtk+-2.18.2.exheres-0 b/packages/x11-libs/gtk+/gtk+-2.18.2.exheres-0
index cb78954..6fe91ea 100644
--- a/packages/x11-libs/gtk+/gtk+-2.18.2.exheres-0
+++ b/packages/x11-libs/gtk+/gtk+-2.18.2.exheres-0
@@ -18,6 +18,7 @@ MYOPTIONS="cups doc
 jpeg2000 [[ description = [ Support for JPEG 2000 Part-1 images using libjasper ] ]]
 jpeg
 man-pages [[ description = [ Build and install man pages for the GTK+ API ] ]]
+nosvg [[ description = [ Work around cairo and svg dependencies for initial install ] ]]
 tiff xinerama
 linguas: ${LANGS}"
 
@@ -37,7 +38,7 @@ DEPENDENCIES="
     build+run:
         dev-libs/atk[>=1.13.0]
         dev-libs/glib:2[>=2.21.3]
-        x11-libs/cairo[>=1.6][X][pdf][svg]
+        x11-libs/cairo[>=1.6][X][pdf] [[ note = [ if you upgrade instead of a fresh install, building cairo with svg makes gtk+ happy ] ]]
         x11-libs/pango[>=1.20][X]
         media-libs/libpng
         media-libs/fontconfig
@@ -67,6 +68,14 @@ DEFAULT_SRC_CONFIGURE_PARAMS=( '--with-xinput' '--with-libpng' '--with-gdktarget
 DEFAULT_SRC_CONFIGURE_OPTION_WITHS=( 'jpeg libjpeg' 'jpeg2000 libjasper' 'tiff libtiff' )
 DEFAULT_SRC_CONFIGURE_OPTION_ENABLES=( 'cups' 'doc gtk-doc' 'man-pages man' xinerama )
 
+src_prepare()
+{
+    if option nosvg; then
+        sed -e "s/SUBDIRS = file/SUBDIRS = /"
+            -i modules/printbackends/Makefile.in || die "Failed to edit Makefile to disable RSVG dependencies"
+    fi
+}
+
 pkg_preinst()
 {
     find "${IMAGE}"/usr/share -depth -type d -empty -delete
-- 
1.6.4.4

